otter/metrics.py

```diff
@@ -253,6 +254,40 @@ def connect_cass_servers(reactor, config):
         seed_endpoints, config['keyspace'], disconnect_on_cancel=True)
 
 
+def log_divergent_groups(clock, divergent_groups, log, timeout, group_metrics):
+    """
+    Log groups that have not changed and been divergent for long time
+    """
+    dg = divergent_groups
+    converged, diverged = partition_bool(
+        lambda gm: gm.actual + gm.pending == gm.desired, group_metrics)
+    # stop tracking all converged groups
+    for gm in converged:
+        dg.pop((gm.tenant_id, gm.group_id), None)
+    # Start tracking divergent groups depending on whether they've changed
+    now = clock.seconds()
+    for gm in diverged:
+        pair = (gm.tenant_id, gm.group_id)
+        if pair in dg:
+            last_time, values = dg[pair]
+            if values != hash((gm.desired, gm.actual, gm.pending)):
+                del dg[pair]
```

Comment:
Non-blocking - if this is deleted, won't it not track this group as divergent until the next metrics cycle?  But this group is currently divergent, it has just moved.  This probably doesn't matter much, since we seem to be gathering metrics once a minute, and the timeout is once an hour.  If the metrics cycle is increased or timeout decreased, it may start to make more of a difference.